clarity/src/vm/docs/mod.rs

```diff
@@ -568,11 +575,133 @@ const XOR_API: SimpleFunctionAPI = SimpleFunctionAPI {
 ",
 };
 
+const BITWISE_XOR_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: None,
+    snippet: "^ ${1:expr-1} ${2:expr-2}",
+    signature: "(^ i1 i2...)",
+    description: "Alias for the `xor` function.  Returns the result of bitwise exclusive or'ing a variable number of integer inputs.",
+    example: "(^ 1 2) ;; Returns 3
+(^ 120 280) ;; Returns 352
+(^ -128 64) ;; Returns -64
+(^ u24 u4) ;; Returns u28
+(^ 1 2 4 -1) ;; Returns -8
+",
+};
+
+const BITWISE_AND_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: Some("& (bitwise and)"),
+    snippet: "& ${1:expr-1} ${2:expr-2}",
+    signature: "(& i1 i2...)",
+    description: "Returns the result of bitwise and'ing a variable number of integer inputs.",
+    example: "(& 24 16) ;; Returns 16
+(& 28 24 -1) ;; Returns 24
+(& u24 u16) ;; Returns u16
+(& -128 -64) ;; Returns -128
+(& 28 24 -1) ;; Returns 24
+"
+};
+
+const BITWISE_OR_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: Some("| (bitwise or)"),
+    snippet: "| ${1:expr-1} ${2:expr-2}",
+    signature: "(| i1 i2...)",
+    description: "Returns the result of bitwise inclusive or'ing a variable number of integer inputs.",
+    example: "(| 4 8) ;; Returns 12
+(| 1 2 4) ;; Returns 7
+(| 64 -32 -16) ;; Returns -16
+(| u2 u4 u32) ;; Returns u38
+"
+};
+
+const BITWISE_NOT_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: Some("~ (bitwise not)"),
+    snippet: "~ ${1:expr-1}",
+    signature: "(~ i1)",
+    description: "Returns the one's compliement (sometimes also called the bitwise compliment or not operator) of `i1`, effectively reversing the bits in `i1`.
+In other words, every bit that is `1` in ì1` will be `0` in the result.  Conversely, every bit that is `0` in `i1` will be `1` in the result.
+",
+    example: "(~ 3) ;; Returns -4
+(~ u128) ;; Returns u340282366920938463463374607431768211327
+(~ 128) ;; Returns -129
+(~ -128) ;; Returns 127
+"
+};
+
+const BITWISE_LEFT_SHIFT_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: Some("<< (binary left shift)"),
+    snippet: "<< ${1:expr-1} ${2:expr-2}",
+    signature: "(<< i1 i2)",
+    description: "Shifts all the bits in `i1` to the left by the number of places specified in `i2`. 
+Excess bits shifted off to the left are discarded, and zero-bits are shifted in from the right.
+
+Yields `i1` << mask(`i2`), where `mask` removes any high-order bits of `i2` that would cause the shift
+to exceed the bitwidth of `i1`.
+
+Note: This is not the same as a rotate-left; The `i2` operand is restricted to the bit range of type of `i1`, 
+rather than the bits shifted off to the left being returned to the other end.
+
+Observe that the second operand (number of positions to shift) must be a 32-bit unsigned integer (meaning
```

Comment:
This is no longer true.  `i2` can be any `u128`; the implementation simply masks out the upper 96 bits.